### PR TITLE
Tunnel mode configurable

### PR DESCRIPTION
**Pull request by Rongo-JL**
Make the tunnel mode be configurable.

b/307998866

### CODE REVIEW

File: starboard/android/shared/player_components_factory.h [hunk 1]
```diff
@@ -318,8 +321,8 @@ class PlayerComponentsFactory : public starboard::shared::starboard::player::
                    << ". Tunnel mode is disabled.";
     }
 
-    if (kForceTunnelMode && !enable_tunnel_mode) {
-      SB_LOG(INFO) << "`kForceTunnelMode` is set to true, force enabling tunnel"
+    if (IsForceTunnelModeEnabled() && !enable_tunnel_mode) {
+      SB_LOG(INFO) << "`force_tunnel_mode` is set to true, force enabling tunnel"
```
Comment (borongc) [inline starboard/android/shared/player_components_factory.h:325]: Format checking: change to
```
      SB_LOG(INFO)
          << "`force_tunnel_mode` is set to true, force enabling tunnel"
          << " mode.";
```

File: starboard/android/shared/player_components_factory.h [hunk 2]
```diff
@@ -195,6 +191,13 @@ class PlayerComponentsFactory : public starboard::shared::starboard::player::
     return (value + alignment - 1) / alignment * alignment;
   }
 
+  static bool IsForceTunnelModeEnabled() {
+    bool force_tunnel_mode =
```
Comment (xiaomings) [inline starboard/android/shared/player_components_factory.h:195]: We'd prefer this to be controlled by the web app.  Do you happen to know if there are any particular devices that we should force enable tunnel mode?  We may consider including the devices in our initial experiment.

Please feel free to continue this discussions via email if that's preferred.

File: starboard/android/shared/media_decoder.cc [hunk 3]
```diff
@@ -317,18 +317,18 @@ void MediaDecoder::DecoderThreadFunc() {
...
-          dequeue_output_results.erase(dequeue_output_results.begin());
+      if (!dequeue_output_results.empty()) {
+        auto& dequeue_output_result = dequeue_output_results.front();
+        if (dequeue_output_result.index < 0) {
+          host_->RefreshOutputFormat(media_codec_bridge_.get());
+        } else {
+          SB_DCHECK(!tunnel_mode_enabled_);
+          host_->ProcessOutputBuffer(media_codec_bridge_.get(),
+                                     dequeue_output_result);
         }
```
Comment (jasonzhangxx) [inline starboard/android/shared/media_decoder.cc:328]: Can you rebase the pr and see if any code here is modified?

File: starboard/android/shared/player_components_factory.h (hunk 1 above)
Comment (Rongo-JL) [reply starboard/android/shared/player_components_factory.h]: Sorry for the incorrect operation that I accidentally closed this PR when I rebase the commit.
I will make a new PR with the correction

File: starboard/android/shared/player_components_factory.h (hunk 2 above)
Comment (Rongo-JL) [reply starboard/android/shared/player_components_factory.h]: Thanks for the notification and review. Sorry for my really late response.
I will discuss internally and let you know through email if we have a device that needs to enable tunnel mode for the experiment.

File: starboard/android/shared/media_decoder.cc (hunk 3 above)
Comment (Rongo-JL) [reply starboard/android/shared/media_decoder.cc]: Thanks for the review.
According to Xiaoming's comment, I think maybe we don't need to enable the tunnel mode forcibly.
If we need to turn tunnel mode on the device for experiment, we will let you know first through email.

**Pull request closed without merging**